docs/05-technical-reference/00-architecture/evnt-01-architecture.md

```diff
@@ -2,30 +2,32 @@
 title: Eventing Architecture
 ---
 
-Eventing uses Event Publisher Proxy and Eventing Controller to connect to the default NATS backend. They work together to process and deliver events in Kyma. See the [Event processing and delivery](../evnt-01-event-processing.md) document for more information.
+Eventing uses Event Publisher Proxy and Eventing Controller to connect to the default NATS backend. They work together to process and deliver events in Kyma. See the [Event types](../evnt-01-event-types.md) document for more information.
 
-## Event Publisher Proxy
+## Event processing and delivery
 
-Event Publisher Proxy receives legacy and Cloud Event publishing requests from the cluster workloads (microservices or Functions). It converts any legacy events to Cloud Events. Then, it redirects events to the NATS server. It also fetches a list of Subscriptions for a connected application.
+The event processing and delivery flow uses the NATS server to process events and send them to subscribers.
+This diagram explains the event flow in Kyma, from the moment the Application sends an event, to the point when the event triggers the Function.
 
-## Eventing Controller
+![Eventing flow](./assets/evnt-architecture.svg)
+
+1. The Eventing Controller watches the Subscription Custom Resource. It determines if there are any new incoming events.
```

Comment:
```suggestion
1. The Eventing Controller watches the Subscription Custom Resource. It detects if there are any new incoming events.
```
I used to love "determine" until I noticed that in translation, it's sometimes interpreted as "detect/feststellen" and sometimes as "specify/bestimmen". Now I try to avoid it because it's ambiguous.